### ISSUE TITLE
Mouse cursor frozen for 250ms after setting absolutePosition

### ISSUE THREAD

**Issue by jameswhqi** [Hammerspoon/hammerspoon#3332]
After calling `hs.mouse.absolutePosition([pos])`, the mouse cursor is frozen for a short period of time, which is undesirable because for my use case I'd like to move my cursor immediately after setting its position. This can be tested with the following code:

```lua
hs.hotkey.bind({'ctrl'}, 'x', function()
  i = 0
  hs.timer.doUntil(function() return i == 10 end, function()
    hs.mouse.absolutePosition(hs.mouse.absolutePosition())
    i = i + 1
  end, 0.25)
end)
```

If you press ctrl+x and keep moving the mouse, the cursor will be frozen for 2.5 seconds. Changing `0.25` to `0.26` will allow a small amount of movement between each call to `absolutePosition()`, so 250ms is approximately the duration that the cursor is frozen.

Versions:
macOS Ventura 13.0.1
Hammerspoon 0.9.97

**Comment by jameswhqi** [comment]
This should have been fixed by #2005; why is it still happening?

**Comment by jameswhqi** [comment]
[This StackOverflow answer](https://stackoverflow.com/a/51787395) seems relevant. Perhaps `CGAssociateMouseAndMouseCursorPosition` only works with earlier versions of macOS while `CGEventSourceSetLocalEventsSuppressionInterval` only works with newer versions?

**Comment by jameswhqi** [comment]
My local test shows that replacing
```objc
    CGWarpMouseCursorPosition(absolutePosition);
    CGAssociateMouseAndMouseCursorPosition(YES);
```
with
```objc
    CGEventSourceRef eventSourceRef = CGEventSourceCreate(kCGEventSourceStateCombinedSessionState);
    CGEventSourceSetLocalEventsSuppressionInterval(eventSourceRef, 0);
    CGWarpMouseCursorPosition(absolutePosition);
```
[here](https://github.com/Hammerspoon/hammerspoon/blob/ac3134075ee5c662ccf3a29f51526ee9ccfcc09d/extensions/mouse/libmouse.m#L108) resolves the issue, although I don't know if this is the right place to put the code (maybe it should be in the object initializer or somewhere even more global?) or how to test it for different macOS versions.

**Comment by cmsj** [comment]
@jameswhqi is this still an issue? Your replacement code makes sense I think, but I'd like to check first.

**Comment by jameswhqi** [comment]
@cmsj Yes, it's still an issue on macOS Sonoma 14.6.1 and Hammerspoon 1.0.0.

**Comment by merlindru** [comment]
Running into this as well with a "middle click to drag" script I found online

Is there any workaround? I know very little about Lua and HS
